File: 2379. Minimum Recolors to Get K Consecutive Black Blocks Easy/solution.js
/**
 * @param {string} blocks
 * @param {number} k
 * @return {number}
 */
 var minimumRecolors = function(blocks, k) {
  let white = 0;
  let result = Infinity;
  for (let index = 0; index < blocks.length; index++){
    let value = blocks[index];
    if (index >= k){
      if (white < result){
        result = white;
      }
    }
    if (value == 'W'){
      white += 1;
    }
    if (index >= k && blocks[index - k] == 'W'){
      white -= 1;
    }
  }

  if (white < result){
    result = white;
  }

  return result;
};

File: 2379. Minimum Recolors to Get K Consecutive Black Blocks Easy/solution.java
class Solution {
  public int minimumRecolors(String blocks, int k) {
    int white = 0;
    double result = Double.POSITIVE_INFINITY;
    for (int index = 0; index < blocks.length(); index++){
      Character value = blocks.charAt(index);
      if (index >= k){
        if (white < result){
          result = white;
        }
      }
      if (value == 'W'){
        white += 1;
      }
      if (index >= k && blocks.charAt(index - k) == 'W'){
        white -= 1;
      }
    }

    if (white < result){
      result = white;
    }

    return (int) result;
  }
}